cmd/cleanup-controller/handlers/cleanup/match.go

```diff
@@ -0,0 +1,254 @@
+package cleanup
+
+import (
+	"fmt"
+	"strings"
+
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	kyvernov2beta1 "github.com/kyverno/kyverno/api/kyverno/v2beta1"
+	"github.com/kyverno/kyverno/pkg/engine/wildcards"
+	"github.com/kyverno/kyverno/pkg/logging"
+	"github.com/kyverno/kyverno/pkg/utils/wildcard"
+	"go.uber.org/multierr"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func checkMatchesResources(
+	resource unstructured.Unstructured,
+	statement kyvernov2beta1.MatchResources,
+	// ruleRef kyvernov1.Rule,
+	//  admissionInfoRef kyvernov1beta1.RequestInfo,
+	dynamicConfig []string,
+	namespaceLabels map[string]string,
+	// policyNamespace string,
+) error {
+	var errs []error
+	if len(statement.Any) > 0 {
+		// include object if ANY of the criteria match
+		// so if one matches then break from loop
+		oneMatched := false
+		for _, rmr := range statement.Any {
+			// if there are no errors it means it was a match
+			if len(checkResourceFilter(
+				rmr,
+				//  admissionInfo,
+				resource,
+				dynamicConfig,
+				namespaceLabels,
+			)) == 0 {
+				oneMatched = true
+				break
+			}
+		}
+		if !oneMatched {
+			errs = append(errs, fmt.Errorf("no resource matched"))
+		}
+	} else if len(statement.All) > 0 {
+		// include object if ALL of the criteria match
+		for _, rmr := range statement.All {
+			errs = append(
+				errs,
+				checkResourceFilter(
+					rmr,
+					// admissionInfo,
+					resource,
+					dynamicConfig,
+					namespaceLabels,
+				)...,
+			)
+		}
+	}
+	return multierr.Combine(errs...)
+}
+
+func checkResourceFilter(
+	statement kyvernov1.ResourceFilter,
+	// admissionInfo kyvernov1beta1.RequestInfo,
```

Comment:
Why comment this out?

Reply:
This happening in the background, we don't have admission infos at hand.
I'm going to split that into other functions.